src/entity/user/utils.ts

```diff
@@ -144,6 +144,47 @@ export const updateUserEmail = async (
   }
 };
 
+export const confirmUserEmail = async (
+  con: DataSource,
+  data: UpdateUserEmailData,
+): Promise<UpdateUserEmailResult> => {
+  if (!data.email || !data.id) {
+    return { status: 'failed', reason: UpdateUserFailErrorKeys.MissingFields };
+  }
+
+  try {
+    const res = await con
+      .getRepository(User)
+      .update({ id: data.id, email: data.email }, { emailConfirmed: true });
+
+    if (res.affected === 0) {
+      logger.info(
+        { id: data.id },
+        'Failed to update email user not found with ID or email',
+      );
+      return {
+        status: 'failed',
+        reason: UpdateUserFailErrorKeys.UserDoesntExist,
+      };
+    }
+
+    logger.info({ id: data.id }, 'Updated email confirmation for user');
```

Comment:
This might spam logs considering amount of users we get.

Reply:
Set it to be debug